pkg/ebpf/tracee.go

```diff
@@ -198,26 +204,37 @@ func (t *Tracee) Stats() *metrics.Stats {
 	return &t.stats
 }
 
-func isEssential(id int32) bool {
-	return EventsDefinitions[id].EssentialEvent
+func (t *Tracee) setEventLoad(id int32) {
+	he := t.events[id]
+	he.load = true
+	t.events[id] = he
+}
+
+func (t *Tracee) setEventSubmit(id int32) {
+	he := t.events[id]
+	he.submit = true
+	t.events[id] = he
 }
 
-func setEssential(id int32) {
-	event := EventsDefinitions[id]
-	event.EssentialEvent = true
-	EventsDefinitions[id] = event
+func (t *Tracee) setEventEmit(id int32) {
+	he := t.events[id]
+	he.emit = true
+	t.events[id] = he
 }
 
-func handleEventsDependencies(e int32, initReq *RequiredInitValues) {
+func (t *Tracee) handleEventsDependencies(e int32, initReq *RequiredInitValues) {
 	eDependencies := EventsDefinitions[e].Dependencies
 	if len(eDependencies.ksymbols) > 0 {
 		initReq.kallsyms = true
 	}
 	// Some events depend on other events - mark those essential
```

Comment:
nit: The semantics of this comment is wrong now since this is marking to load and possibly submit, not to be essential.

Reply:
Right.
For a matter of fact, we can remove this comment since this logic is under a function named hanleEventDependencies.